pkg/services/meta/block.go

```diff
@@ -0,0 +1,94 @@
+package meta
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/nspcc-dev/neo-go/pkg/core/block"
+	"github.com/nspcc-dev/neo-go/pkg/core/mpt"
+	"github.com/nspcc-dev/neo-go/pkg/neorpc"
+	"go.uber.org/zap"
+)
+
+func (m *Meta) blockFetcher(ctx context.Context, buff <-chan *block.Header) {
```

Comment:
could u pls move these methods to a separate source file in a separate patch? this would easify the review

Reply:
done

Reply:
btw, i cost me additional 30+ mins for rebasing, conflicting, fixing, and rechecking additional threads below. this PR changes logic in general so old functions become separate things (it is by blocks now), but still PR is 200- lines, imo it was not worth it but it is already done

Reply:
ive no idea how u could face any conflicts with this change, but thx anyway

Reply:
because i was moving parts that should be changed to fix other threads. change it once in the original code, and change it second time in the moving commit. moving commit cannot be first because there was no "block" logic before and imo it was a good change as a single commit that changed everything